app/celery/tasks.py

```diff
@@ -0,0 +1,57 @@
+import base64
+
+from botocore.exceptions import ClientError as BotoClientError
+from flask import current_app
+from notifications_utils.s3 import s3download, s3upload
+from notifications_utils.statsd_decorators import statsd
+
+from app import notify_celery, TaskNames, QueueNames
+from app.precompiled import sanitise_file_contents
+
+
+@notify_celery.task(name='sanitise-and-upload-letter')
+@statsd(namespace='template-preview')
+def sanitise_and_upload_letter(notification_id, filename):
+    current_app.logger.info('Sanitising notification with id {}'.format(notification_id))
+
+    try:
+        pdf_content = s3download(current_app.config['LETTERS_SCAN_BUCKET_NAME'], filename).read()
+        json_response = sanitise_file_contents(pdf_content)
+
+        # Only files that have failed validation have a message in the json_response
```

Comment:
Discussed maybe changing comment to be consistent around `sanitisation` rather than `validation`